internal/oci/container.go

```diff
@@ -344,6 +350,104 @@ func (c *Container) exitFilePath() string {
 	return filepath.Join(c.dir, "exit")
 }
 
+// IsRunning is a function that checks if a specific pid is running
+// it is used to check a container state when we don't want (or don't trust) a `$runtime state` call
+func (c *Container) IsRunning() bool {
+	_, err := c.pid()
+	if err != nil {
+		logrus.Errorf("checking if pid of %s is running failed: %v", c.id, err)
+		return false
+	}
+
+	return true
+}
+
+// Pid is the external, locking function that returns the container's
+// pid, if it's running, and actually the container process
+func (c *Container) Pid() (int, error) {
+	c.opLock.Lock()
+	defer c.opLock.Unlock()
+	return c.pid()
+}
+
+// pid is the internal, non-locking version of pid. It checks
+// that we have a Pid defined in the state, that pid can be found
+// and it is the same process that was originally started by the runtime
+func (c *Container) pid() (int, error) {
+	if c.state == nil {
+		return -1, errors.New("state not initialized")
+	}
+	pid := c.state.Pid
+	if pid <= 0 {
+		return -1, errors.New("pid not initialized")
```

Comment:
We can introduce an error variable here like `ErrPidNotRunning`

Reply:
I'm not sure we need that. I'd be fine doing it but there's no place we really want to check against this as of now